### PR TITLE
fix: handle float-formatted IDs from ArmA 3 extension API

### PR DESCRIPTION
**Pull request by fank**
## Summary
- ArmA 3's SQF has no integer type, so the extension API may serialize IDs as floats (e.g. `"32.00"` instead of `"32"`), causing `strconv.ParseUint` failures in `:NEW:VEHICLE:STATE:` and other handlers
- Added `parseUintFromFloat` / `parseIntFromFloat` helpers that try integer parsing first (fast path), then fall back to `ParseFloat`
- Replaced all 17 ID parsing call sites across soldier, vehicle, projectile, hit, kill, chat, radio, and ACE3 event handlers

## Test plan
- [x] Unit tests for `parseUintFromFloat` (10 cases: integers, floats, edge cases)
- [x] Unit tests for `parseIntFromFloat` (9 cases: integers, negatives, floats, edge cases)
- [x] End-to-end test for `LogVehicleState` with float ocapId `"32.00"`
- [x] End-to-end test for `LogSoldierState` with float ocapId `"30.00"`
- [x] Full test suite passes

### CODE REVIEW

File: internal/handlers/handlers.go [hunk 1]
```diff
@@ -24,6 +24,31 @@ import (
...
+func parseUintFromFloat(s string) (uint64, error) {
+	if v, err := strconv.ParseUint(s, 10, 64); err == nil {
+		return v, nil
+	}
+	f, err := strconv.ParseFloat(s, 64)
+	if err != nil {
+		return 0, fmt.Errorf("strconv.ParseUint/Float: parsing %q: invalid syntax", s)
+	}
+	return uint64(f), nil
+}
```
[the remark this answers is not in the document]
Comment (fank) [reply internal/handlers/handlers.go:41]: Fixed. Added `f < 0` and `f != float64(uint64(f))` checks, and now returning the original `ParseFloat` error instead of a hardcoded message.

File: internal/handlers/handlers.go [hunk 2]
```diff
@@ -24,6 +24,31 @@ import (
...
+func parseIntFromFloat(s string) (int64, error) {
+	if v, err := strconv.ParseInt(s, 10, 64); err == nil {
+		return v, nil
+	}
+	f, err := strconv.ParseFloat(s, 64)
+	if err != nil {
+		return 0, fmt.Errorf("strconv.ParseInt/Float: parsing %q: invalid syntax", s)
+	}
+	return int64(f), nil
+}
```
[the remark this answers is not in the document]
Comment (fank) [reply internal/handlers/handlers.go:56]: Fixed. Added `f != float64(int64(f))` check.

File: internal/handlers/handlers_test.go [hunk 3]
```diff
@@ -329,6 +329,136 @@ func TestLogSoldierState_ScoresParsingPanic(t *testing.T) {
...
+		{"integer", "32", 32, false},
+		{"zero", "0", 0, false},
+		{"float with decimals", "32.00", 32, false},
+		{"float with trailing zero", "30.0", 30, false},
+		{"float truncates", "10.99", 10, false},
+		{"large integer", "65535", 65535, false},
+		{"large float", "65535.00", 65535, false},
+		{"empty string", "", 0, true},
+		{"non-numeric", "abc", 0, true},
+		{"negative wraps", "-1", ^uint64(0), false}, // float-to-uint wraps; callers use parseIntFromFloat for signed IDs
```
[the remark this answers is not in the document]
Comment (fank) [reply internal/handlers/handlers_test.go]: Fixed. Negative now expects error, fractional (`"10.99"`) also expects error in both helpers.
